man/man8/zfs.8

```diff
@@ -1010,6 +1057,68 @@ This property can also be referred to by its shortened column name,
 
 .sp
 .ne 2
+.na
+\fB\fBencryption\fR=\fBoff\fR | \fBon\fR | \fBaes-128-ccm\fR |
+\fBaes-192-ccm\fR | \fBaes-256-ccm\fR | \fBaes-128-gcm\fR |
+\fBaes-192-gcm\fR | \fBaes-256-gcm\fR
+.ad
+.sp .6
+.RS 4n
+Controls the encryption cipher suite (block cipher, key length, and mode) used
+for this dataset. Requires the encryption feature to be enabled on the pool.
+Requires a \fBkeyformat\fR to be set at dataset creation time.
+.sp
+Selecting \fBencryption\fR=\fBon\fR when creating a dataset indicates that the
+default encryption suite will be selected, which is currently \fBaes-256-ccm\fR.
+In order to provide consistent data protection, encryption must be specified at
+dataset creation time and it cannot be changed afterwards.
+.sp
+For more details and caveats about encryption see the "Encryption" section.
+.RE
+
+.sp
+.ne 2
+.mk
+.na
+\fB\fBkeyformat\fR=\fBraw\fR | \fBhex\fR | \fBpassphrase\fR
+.ad
+.sp .6
+.RS 4n
+Controls what format the user's encryption key will be provided as. This
+property is only set when the dataset is encrypted.
+.sp
+Raw keys and hex keys must be 32 bytes long (regardless of the chosen
+encryption suite) and must be randomly generated. Passphrases must be between
```

Comment:
Hmmm. On second thought, after double checking my notes, I remember a lot of controversy over this which I'm not sure we really want to be a part of (mostly `/dev/random` vs `/dev/urandom`). Its tricky because the standard manual pages suggest one thing but there are many contrary opinions otherwise. I will leave this as is for now, (unless you have a strong opinion one way or another).